oh_staff_ui/views_utils.py

```diff
@@ -466,3 +470,50 @@ def run_process_file_command(
     # testing confirms this, though it does close after 30 seconds or so.
     # To be safe, explicitly close this one when done.
     connection.close()
+
+
+def get_listrecords_oai(verb: str, ark: str = None) -> str:
+
+    pi_set = ProjectItem.objects.filter(status__status__iexact="completed")
+
+    if ark:
+        ark = ark.replace("-", "/")
+        pi_set = pi_set.filter(ark=ark)
+
+    verb_element = etree.Element(verb)
+
+    for pi in pi_set:
+        verb_element.append(OralHistoryMods(pi).add_oai_envelope())
+
+    return wrap_oai_content(verb_element, verb, ark)
+
+
+def get_record_oai(ark: str = None) -> str:
+    return get_listrecords_oai("GetRecord", ark)
+
+
+def wrap_oai_content(xml_element, verb: str, ark_ns: str) -> str:
+
+    oai_envelope = """
+            <OAI-PMH xmlns="http://www.openarchives.org/OAI/2.0/" 
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+                xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/ 
+                http://www.openarchives.org/OAI/2.0/OAI-PMH.xsd" />
+            """
+    oai_tree = etree.fromstring(oai_envelope)
+    d = datetime.now()
+
+    etree.SubElement(oai_tree, "responseDate").text = d.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    req = f"""<request metadataPrefix="mods" 
```

Comment:
Please add comments explaining this.  I think the `reverse` here is a Django URL shortcut, so... is some sort of URL being built here?